src/shapes/image.class.js

```diff
@@ -121,15 +122,27 @@
      * @chainable
      */
     setElement: function(element, callback, options) {
+
+      var _callback, _this;
+
       this._element = element;
       this._originalElement = element;
       this._initConfig(options);
 
+      if (this.resizeFilters.length === 0) {
+        _callback = callback;
+      } else {
+        _this = this;
+        _callback = function() {
+          _this.applyFilters(callback, _this.resizeFilters, _this._filteredEl || _this._originalElement, true);
+        }
+      }
```

Comment:
if there are resize filters in the restore from object/json, let's fire the callback after we restore also those.